### ISSUE TITLE
Error when adding widget path

### ISSUE THREAD

**Issue by newf276** [louisbraun/script.unfussy.helper#1]
Adding widget path on matrix for either seren or a4kstreaming adding return error:

2021-02-14 07:25:29.146 T:83286    INFO <general>: Loading skin file: DialogContextMenu.xml, load type: KEEP_IN_MEMORY
2021-02-14 07:25:31.599 T:83811    INFO <general>: initializing python engine.
2021-02-14 07:25:31.644 T:83811   ERROR <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'AttributeError'>
                                                   Error Contents: 'xbmcgui.ListItem' object has no attribute 'getfilename'
                                                   Traceback (most recent call last):
                                                     File "/Users/ian/Library/Application Support/Kodi/addons/script.unfussy.helper/contextitem.py", line 5, in <module>
                                                       widget_path = sys.listitem.getfilename()
                                                   AttributeError: 'xbmcgui.ListItem' object has no attribute 'getfilename'
                                                   -->End of Python script error report<--
                                                   
2021-02-14 07:25:31.791 T:83811    INFO <general>: Python interpreter stopped
2021-02-14 07:26:09.858 T:83421 WARNING <general>: DARWINOSX: underflow (0 vs 4096 bytes)
2021-02-14 07:26:45.233 T:85292    INFO <general>: initializing python engine.
2021-02-14 07:26:45.442 T:85292 WARNING <general>: xbmc.translatePath is deprecated and might be removed in future kodi versions. Please use xbmcvfs.translatePath instead.
2021-02-14 07:26:45.608 T:85292    INFO <general>: Skipped 16 duplicate messages..
2021-02-14 07:26:45.608 T:85292    INFO <general>: SEREN: Seren, Running Path - {}
2021-02-14 07:26:45.706 T:85292    INFO <general>: CPythonInvoker(10, /Users/ian/Library/Application Support/Kodi/addons/plugin.video.seren/seren.py): script successfully run
2021-02-14 07:26:45.706 T:85292    INFO <general>: CPythonInvoker(10, /Users/ian/Library/Application Support/Kodi/addons/plugin.video.seren/seren.py): waiting on thread 123145446666240
2021-02-14 07:26:48.800 T:83286   ERROR <general>: Control 55 in window 10025 has been asked to focus, but it can't
2021-02-14 07:26:48.813 T:85421    INFO <general>: initializing python engine.
2021-02-14 07:26:48.974 T:85292 WARNING <general>: CPythonInvoker(10, /Users/ian/Library/Application Support/Kodi/addons/plugin.video.seren/seren.py): the python script "/Users/ian/Library/Application Support/Kodi/addons/plugin.video.seren/seren.py" has left several classes in memory that we couldn't clean up. The classes include: N9XBMCAddon4xbmc8PlayListE,N9XBMCAddon9xbmcaddon5AddonE,N9XBMCAddon4xbmc7MonitorE,N9XBMCAddon7xbmcgui6WindowE
2021-02-14 07:26:48.975 T:85292    INFO <general>: Python interpreter stopped
2021-02-14 07:26:49.400 T:85421    INFO <general>: CPythonInvoker(11, /Users/ian/Library/Application Support/Kodi/addons/plugin.video.a4kstreaming/main.py): script successfully run
2021-02-14 07:26:52.708 T:85503    INFO <general>: initializing python engine.
2021-02-14 07:26:52.767 T:85503   ERROR <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'AttributeError'>
                                                   Error Contents: 'xbmcgui.ListItem' object has no attribute 'getfilename'
                                                   Traceback (most recent call last):
                                                     File "/Users/ian/Library/Application Support/Kodi/addons/script.unfussy.helper/contextitem.py", line 5, in <module>
                                                       widget_path = sys.listitem.getfilename()
                                                   AttributeError: 'xbmcgui.ListItem' object has no attribute 'getfilename'
                                                   -->End of Python script error report<--
                                                   
2021-02-14 07:26:52.947 T:85503    INFO <general>: Python interpreter stopped

**Comment by louisbraun** [comment]
Fixed in kodi19-matrix branch.
Thx for reporting.

Cheers Louis

**Comment by newf276** [comment]
> Fixed in kodi19-matrix branch.
> Thx for reporting.
> 
> Cheers Louis

I will test and let you know, thanks for the quick response.  I am trying to have a small build, with only A4kStreaming for IMD watchlist, and Seren with my Movies/TV Shows Collection.  Because your skin is fast and simple, I am hoping to get it working with widgets form Seren and A4K.  Thanks will report back soon.

**Comment by newf276** [comment]
> > Fixed in kodi19-matrix branch.
> > Thx for reporting.
> > Cheers Louis
> 
> I will test and let you know, thanks for the quick response. I am trying to have a small build, with only A4kStreaming for IMD watchlist, and Seren with my Movies/TV Shows Collection. Because your skin is fast and simple, I am hoping to get it working with widgets form Seren and A4K. Thanks will report back soon.

Error has been fixed, it creates the widgets as expected, and I was able to populate the homepage with the ones I created.  Only issue I find is that when selecting an actual show, vice an episode, it does nothing. In order to browse the show seasons, I have to select c or menu command, and select browse to get to the actual seasons. While this is not a concern for next episodes it actually plays the episode or a movie when selected form a movie widget.  IS there a way for Shows when listed on the widget home screen to actually browse to the seasons? Or is there a way perhaps I can actually edit the widgets it creates in the unfussy data folder to edit the code that points to the widget. For example edit: {"id": 1, "name": "Seren My Movies", "path": "plugin://plugin.video.seren/?action=moviesMyCollection&from_widget=true"}

Thanks ....
